### PR TITLE
Modernize build (Require Java 8 & Jenkins 2.60.3)

### PR DESCRIPTION
**Pull request by TobiX**
Update to Jenkins 2.60.3 (released 2017-08-17) and Java 8

Considerations:
- From [Telemetry](http://stats.jenkins.io/pluginversions/lockable-resources.html) we can see, that this only "affects" 553/112545 (0.5%) of users
- If we only consider the last 4 releases (2.1-2.4), this affects 185/111776 (just under 0.2%) of "updating" users

Changes:
- Update plugin parent (replaces FindBugs with SpotBugs)
- Doesn't use the pipeline aggregator anymore, but individual dependencies
- Re-enables SpotBugs (and fix the one place it found)
- Build against recommended Jenkins/JDK versions (not sure if all combinations work, a member of this project needs to push my branch to make Jenkins pick up the Jenkinsfile change)
- Plugin dependencies are updated, but still older then the required Jenkins baseline...

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -30,12 +30,9 @@
 	<url>https://wiki.jenkins-ci.org/display/JENKINS/Lockable+Resources+Plugin</url>
```
Comment (darxriggs) [inline pom.xml]: This could be updated to the newer "https://wiki.jenkins.io/display/JENKINS/Lockable+Resources+Plugin".

**Comment by darxriggs** [comment]
Upgrading the Jenkins dependency to 2.60.3 here is also very useful for pull request #122 which requires this for implementing a JCasC test.

**Comment by darxriggs** [comment]
@haithir / @fbelzunc / @dcendents / @amuniz / @abayer who is really maintaining this plugin at the moment? Can someone please have a look at this pull request?

File: pom.xml (hunk 1 above)
Comment (TobiX) [reply pom.xml]: Right, done.

**Comment by TobiX** [comment]
@haithir Are you the current maintainer of this plugin and can take a look?

**Comment by fbelzunc** [comment]
I am not the maintainer of this plugin :-)

**Pull request merged**